src/logging/metrics-logger.ts

```diff
@@ -0,0 +1,112 @@
+import {hrtime} from 'process';
+import {Event} from '../event.js';
+import {Logger} from './logger.js';
+
+interface Metric {
+  name: string;
+  matches: (event: Event) => boolean;
+  count: number;
+}
+
+/**
+ * A {@link Logger} that keeps track of metrics.
+ */
+export class MetricsLogger implements Logger {
+  private readonly _actualLogger: Logger;
+  private _startTime: [number, number] = hrtime();
+  private readonly _metrics: Metric[] = [
+    {
+      name: 'Success',
+      matches: (e: Event) => e.type === 'success',
+      count: 0,
+    },
+    {
+      name: 'Ran',
+      matches: (e: Event) => e.type === 'success' && e.reason === 'exit-zero',
+      count: 0,
+    },
+    {
+      name: 'Skipped (fresh)',
```

Comment:
I don't think it'll be very clear from a user's perspective how`Skipped (cached)` is different from `Skipped (fresh)`. Consider instead `Restored from cache`

Reply:
Agreed - changing to `Restored from cache`